veracruz-client/src/cli.rs

```diff
@@ -0,0 +1,301 @@
+    qprintln!(opt, "Connecting to {}", policy.veracruz_server_url());
+    let mut veracruz_client = match VeracruzClient::with_policy_and_hash(
+        opt.identity,
+        opt.key,
+        policy.clone(),
+        policy_hash,
+        &opt.target,
+    ) {
+        Ok(veracruz_client) => veracruz_client,
+        Err(err) => {
+            eprintln!("{}", err);
+            process::exit(1);
+        }
+    };
+
+    let mut did_something = false;
+
+    // send program(s)?
+    for (program_name, program_path) in opt.program.iter().flatten() {
+        qprintln!(opt, "Submitting <enclave>/{} from {}",
+            program_name,
+            match program_path.to_string_lossy().as_ref() {
+                "-" => "<stdout>",
+                path => path,
+            });
+        did_something = true;
+
+        let program_data = if program_path == &path::PathBuf::from("-") {
+            let mut program_data = Vec::new();
+            match io::stdin().read_to_end(&mut program_data) {
+                Ok(_) => program_data,
+                Err(err) => {
+                    eprintln!("{}", err);
+                    process::exit(1);
+                }
+            }
+        } else {
+            match fs::read(program_path) {
+                Ok(program_data) => program_data,
+                Err(err) => {
+                    eprintln!("{}", err);
+                    process::exit(1);
+                }
+            }
+        };
+
+        match veracruz_client.send_program(&program_name, &program_data) {
+            Ok(()) => {}
+            Err(err) => {
+                eprintln!("{}", err);
+                process::exit(1);
+            }
+        }
+    }
+
+    // send data(s)?
+    for (data_name, data_path) in opt.data.iter().flatten() {
+        qprintln!(opt, "Submitting <enclave>/{} from {}",
+            data_name,
+            match data_path.to_string_lossy().as_ref() {
+                "-" => "<stdout>",
+                path => path,
+            });
+        did_something = true;
+
+        let data_data = if data_path == &path::PathBuf::from("-") {
+            let mut data_data = Vec::new();
+            match io::stdin().read_to_end(&mut data_data) {
+                Ok(_) => data_data,
+                Err(err) => {
+                    eprintln!("{}", err);
+                    process::exit(1);
+                }
+            }
+        } else {
+            match fs::read(data_path) {
+                Ok(data_data) => data_data,
+                Err(err) => {
+                    eprintln!("{}", err);
+                    process::exit(1);
+                }
+            }
+        };
+
+        match veracruz_client.send_data(data_name, &data_data) {
+            Ok(()) => {}
+            Err(err) => {
+                eprintln!("{}", err);
+                process::exit(1);
+            }
+        }
+    }
+
+    // fetch result(s)?
+    // TODO why does results take the path to the _binary_? can this
```

Comment:
Is this TODO still relevant?  If so, can you add it to the issue tracker?

Reply:
Yes it is still relevant and may be too much of a change for the PR, can raise an issue

Reply:
Was an issue raised here?

Reply:
Nope! Good catch, I almost missed this:
https://github.com/veracruz-project/veracruz/issues/175